src/main/scala/org/broadinstitute/dsde/workbench/avram/service/CollectionsService.scala

```diff
@@ -1,25 +1,26 @@
 package org.broadinstitute.dsde.workbench.avram.service
 
 import java.util.UUID
-import javax.servlet.http.HttpServletResponse
 
+import javax.servlet.http.HttpServletResponse
 import org.broadinstitute.dsde.workbench.avram.model.{AvramException, Collection, SamResource}
+import org.broadinstitute.dsde.workbench.avram.util.AvramResult
+
 import scala.concurrent.{ExecutionContext, Future}
 
 
 class CollectionsService(implicit executionContext: ExecutionContext) extends AvramService {
 
-  def createCollection(samResource: SamResource, createdBy: String): Future[Either[AvramException, Collection]] = {
-    database.inTransaction { dataAccess =>
-      val collectionExternalId = UUID.randomUUID()
-      dataAccess.collectionQuery.save(collectionExternalId, samResource, createdBy)
-    } map(Right(_))
+  def createCollection(samResource: SamResource, createdBy: String): AvramResult[Collection] = {
+    val collectionExternalId = UUID.randomUUID()
+    AvramResult.fromFuture(database.inTransaction(_.collectionQuery.save(collectionExternalId, samResource, createdBy)))
   }
 
-  def getCollection(externalId: UUID): Future[Either[AvramException, Collection]] = {
-    database.inTransaction { dataAccess =>
-      dataAccess.collectionQuery.getCollectionByExternalId(externalId)
-    } map (_.toRight(AvramException(HttpServletResponse.SC_NOT_FOUND, s"Collection ${externalId.toString} not found")))
+  def getCollection(externalId: UUID): AvramResult[Collection] = {
+    for {
+      result <- AvramResult.fromFuture(database.inTransaction(_.collectionQuery.getCollectionByExternalId(externalId)))
```

Comment:
Minor: I wonder if `database.inTransaction` could be made to return an `AvramResult` directly so we don't always have wrap it in`fromFuture`.